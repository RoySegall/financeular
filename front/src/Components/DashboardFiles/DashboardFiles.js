import IllustrationWithMessage from "../IllustrationWithMessage/IllustrationWithMessage";
import {Link} from "react-router-dom";
import noFiles from "../../Pages/Dashboard/noFiles.svg";
import CardTable from "../Table/CardTable";
import {SmallButton} from "../Buttons/Buttons";
import React from "react";

export const DashboardFiles = ({data}) => {

  const {files} = data.me;

  if (files.length === 0) {
    return <IllustrationWithMessage
      message="No files were found...."
      secondMessage={<>We could not found files. You can <Link to="/upload">upload</Link> some files and get some interesting insights,
        maybe... We can't promise  any thing.</>}
      image={noFiles}
      imageAlt="No files were found"
    />
  }

  return <CardTable
    title={<div className="flex justify-between">
      <div>Uploaded files</div>
      <div>
        <Link to="/upload" className="no-underline"><SmallButton color='green'>Upload another file</SmallButton></Link>
      </div>
    </div>}
    headers={['File name', 'Created at', 'Actions']}
    rows={files.map(file => [
      file.name,
      file.created_at,
      <ul className="flex">
        <li className="pr-4"><Link to={`/dashboard/file/view/${file.id}`}>View</Link></li>
        <li>Delete</li>
      </ul>
    ])}
  />
}
