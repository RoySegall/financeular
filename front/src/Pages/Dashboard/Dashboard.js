import React from 'react';
import "./Dashboard.scss"
import PageTitle from "../../Components/PageTitle/PageTitle";
import NoFiles from "../../Components/NoFiles/NoFiles";
import {useQuery} from "@apollo/client";
import {FILES} from "../../Apollo/Files";
import loadingImage from "./loading.svg";

const DashboardFiles = ({loading, data}) => {

  if (loading) {
    return <img src={loadingImage} alt={"loading"} />
  }

  const {files} = data;

  if (files.length === 0) {
    return <NoFiles />;
  }

  return 'Files';
}

export default () => {
  const { loading, data } = useQuery(FILES);

  return <div className="files-wrapper">
    <PageTitle align='left'>Your files</PageTitle>
    <div>
      <DashboardFiles loading={loading} data={data} />
    </div>
  </div>
}
