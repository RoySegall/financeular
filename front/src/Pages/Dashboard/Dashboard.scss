.files-wrapper {
  padding-left: 1.25em;
  text-align: center;
  width: 100vw;

  .no-files {
    margin: auto;
    width: 25vw;
    padding-top: .5em;
  }
}
