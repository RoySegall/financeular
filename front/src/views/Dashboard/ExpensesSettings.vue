<template>
    <div class="setting">
        <form>
            <div class="form-group">
                <label for="exampleInputEmail1">Add new expenses</label>
                <input type="text" class="form-control" id="exampleInputEmail1" aria-describedby="emailHelp"
                       placeholder="">
            </div>
            <button type="submit" class="btn btn-primary"><i class="fal fa-save"></i> Submit</button>
        </form>

        <hr/>

        <table class="table table-striped">
            <thead>
            <tr>
                <th scope="col">Income</th>
                <th scope="col">Valid since</th>
                <th scope="col">Work place</th>
                <th scope="col">Options</th>
            </tr>
            </thead>
            <tbody>
            <tr>
                <td>15,000</td>
                <td>June 2018</td>
                <td>Coffe des chat</td>
                <td>Set as default | Remove</td>
            </tr>
            <tr>
                <td>6,000</td>
                <td>June 2010 - May 2011</td>
                <td>Pizza la tom</td>
                <td>Set as default | Remove</td>
            </tr>
            </tbody>
        </table>    </div>
</template>

<style lang="scss">

</style>

<script language="ts">
    import {Component, Vue} from 'vue-property-decorator';

    @Component({})
    export default class ExpensesSettings extends Vue {
    }
</script>