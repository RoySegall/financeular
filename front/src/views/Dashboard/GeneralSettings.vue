<template>
    <div class="setting">
        General settings
    </div>
</template>

<style lang="scss">

</style>

<script language="ts">
    import {Component, Vue} from 'vue-property-decorator';

    @Component({})
    export default class GeneralSettings extends Vue {
    }
</script>