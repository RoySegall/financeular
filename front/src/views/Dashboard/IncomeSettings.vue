<template>
    <div class="setting">
        Manage incomes
    </div>
</template>

<style lang="scss">

</style>

<script language="ts">
    import {Component, Vue} from 'vue-property-decorator';

    @Component({})
    export default class IncomeSettings extends Vue {
    }
</script>