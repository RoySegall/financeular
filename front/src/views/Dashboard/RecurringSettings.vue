<template>
    <div class="setting">
        Recurring payments
    </div>
</template>

<style lang="scss">

</style>

<script language="ts">
    import {Component, Vue} from 'vue-property-decorator';

    @Component({})
    export default class RecurringSettings extends Vue {
    }
</script>