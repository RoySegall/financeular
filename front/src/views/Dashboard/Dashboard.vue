<template>
    <div>
        <h1 class="text-left">Dashboard</h1>

        <hr/>

        <div class="row settings-wrapper">
            <div class="menu col-md-2">
                <ul>
                    <li v-for="(menu, key) in menu_items" v-bind:class="{selected: active === key, first: key === 'general', last: key === 'expenses'}" v-html="menu" @click="changeActiveMenu(key)"></li>
                    <li><router-link to="/"><i class="fal fa-hand-point-left"></i> Go back</router-link></li>
                </ul>
            </div>

            <div class="page col-md-10">
                <GeneralSettings v-if="active === 'general' "></GeneralSettings>
                <RecurringSettings v-if="active === 'payments' "></RecurringSettings>
                <IncomeSettings v-if="active === 'incomes' "></IncomeSettings>
                <ExpensesSettings v-if="active === 'expenses' "></ExpensesSettings>
            </div>
        </div>
    </div>

</template>

<style lang="scss">

    .settings-wrapper {
        margin: 0;
        border: $blue1 solid 1px;
        min-height: 75vh;

        .page {
            padding: 1em;
            background: $background-blue;
            text-align: left;

            .setting {
                background: white;
                padding: 1em;
                height: 100%;
                border: $blue1 solid 1px;
            }
        }

        .menu {
            padding: 0;
            margin: 0;
            background: white;
            text-align: left;

            ul {
                margin: 0;
                padding: 0;

                li {
                    list-style-type: none;
                    font-size: 1.25em;
                    padding: .5em 1em;

                    &:hover {
                        cursor: pointer;
                        color: $blue7;
                    }

                    &.selected {
                        background: $background-blue;
                        color: $blue6;
                    }

                    &.first {
                        border-top: none;
                    }

                }
            }
        }
    }

</style>

<script lang="ts">
    import {Component, Vue} from 'vue-property-decorator';
    import GeneralSettings from './GeneralSettings.vue';
    import RecurringSettings from './RecurringSettings.vue';
    import IncomeSettings from './IncomeSettings.vue';
    import ExpensesSettings from './ExpensesSettings.vue';

    @Component({
        components : { GeneralSettings, RecurringSettings, IncomeSettings, ExpensesSettings }
    })

    export default class Dashboard extends Vue {

        private menu_items: object;
        private active: string = 'general';

        public data() {
            return {
                menu_items: {
                    general: '<i class="fal fa-cog"></i> General settings',
                    payments: '<i class="fal fa-recycle"></i> Recurring payments',
                    incomes: '<i class="fal fa-credit-card"></i> Manage incomes',
                    expenses: '<i class="fal fa-hands-usd"></i> Default expenses',
                },
                active: 'general',
            };
        }

        public changeActiveMenu(key) {
            this.active = key;
        }
    }

</script>
