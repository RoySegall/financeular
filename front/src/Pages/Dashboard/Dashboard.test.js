import React from 'react';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {elementShouldContainText, mountComponent} from "../../Tests/Utils";
import renderer from 'react-test-renderer';
import {DashboardFiles} from "../../Components/DashboardFiles/DashboardFiles";
import Dashboard from "./Dashboard";

configure({adapter: new Adapter()});

const mockRedirectLogin = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    Redirect: ({ to }) => mockRedirectLogin,
    Link: ({children}) => children
  };
});

describe('Dashboard component', () => {

  it('Should return the "Loading component" when loading the data', async () => {
    const wrapper = mountComponent({component: <Dashboard loading={true} />, wrapWithProvider: true});

    elementShouldContainText(wrapper, 'Loading');
    elementShouldContainText(wrapper, 'We are working on getting data from the server');
  });

  it('Should return the no files component when the response is empty', () => {
    const wrapper = mountComponent({component: <DashboardFiles loading={false} data={{me: {files: []}}} />, wrapWithProvider: false});

    elementShouldContainText(wrapper, 'No files were found....');
    elementShouldContainText(wrapper, 'We could not found files. You can upload some files');
  });

  it('Should show the list of files when querying the server', () => {
    const tree = renderer
      .create(<DashboardFiles
        loading={false}
        data={{me: {files: [{id: 42, name: 'john', created_at: 'yesterday'}]}}}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
