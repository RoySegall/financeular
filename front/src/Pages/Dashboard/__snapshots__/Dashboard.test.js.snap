// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Should show the list of files when querying the server 1`] = `
<div
  className="relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded bg-white"
>
  <div
    className="rounded-t mb-0 px-4 py-3 border-0"
  >
    <div
      className="flex flex-wrap items-center"
    >
      <div
        className="relative w-full px-4 max-w-full flex-grow flex-1 text-left"
      >
        <h3
          className="font-semibold text-lg text-gray-800"
        >
          <div
            className="flex justify-between"
          >
            <div>
              Uploaded files
            </div>
            <div>
              <button
                className="bg-green-500 text-white active:bg-green-600 font-bold uppercase text-xs px-4 py-2 rounded shadow hover:shadow-md outline-none focus:outline-none mr-1 ease-linear transition-all duration-150"
              >
                Upload another file
              </button>
            </div>
          </div>
        </h3>
      </div>
    </div>
  </div>
  <div
    className="block w-full overflow-x-auto"
  >
    <table
      className="items-center w-full bg-transparent border-collapse"
    >
      <thead>
        <tr>
          <th
            className="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-no-wrap font-semibold text-left bg-gray-100 text-gray-600 border-gray-200"
            id={0}
          >
            File name
          </th>
          <th
            className="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-no-wrap font-semibold text-left bg-gray-100 text-gray-600 border-gray-200"
            id={1}
          >
            Created at
          </th>
          <th
            className="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-no-wrap font-semibold text-left bg-gray-100 text-gray-600 border-gray-200"
            id={2}
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
            id={0}
          >
            john
             
          </td>
          <td
            className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
            id={1}
          >
            yesterday
             
          </td>
          <td
            className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
            id={2}
          >
            <ul
              className="flex"
            >
              <li
                className="pr-4"
              >
                View
              </li>
               
              <li>
                Delete
              </li>
            </ul>
             
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
